internal/pkg/service/buffer/worker/upload/upload_test.go

```diff
@@ -90,19 +89,21 @@ func TestSliceUploadTask(t *testing.T) {
 	// Get slices
 	emptySlice, err := str.GetSlice(ctx, emptySliceKey)
 	assert.NoError(t, err)
-	notEmptySlice, err := str.GetSlice(ctx, notEmptySliceKey)
+	slice1, err := str.GetSlice(ctx, slice1Key)
 	assert.NoError(t, err)
 
-	// Switch slices to the closing state
+	// Switch both slices to the "closing" state and create new slices.
 	clk.Add(10 * time.Second)
-	assert.NoError(t, str.SetSliceState(ctx, &emptySlice, slicestate.Closing))
+	_, err = str.SwapSlice(ctx, &emptySlice)
+	assert.NoError(t, err)
```

Comment:
Namiesto toho, aby sa iba zavrela slice, tak sa aj vytvori nova, ako to ma byt spravne (respektive ako to zabezpecuju ostatne casti apky).